### DISCUSSION TITLE
Access request for ChallengerLTE.cpp board variant branch

### DISCUSSION THREAD

**Discussion by Hoxolotl** [earlephilhower/arduino-pico#1045] [General]
Hello there,

I'm about to edit the ChallengerLTE.cpp variant locally for a Invector Labs Challenger RP2040 LTE board. To add different classes/functions with AT commands I need to send UDP data over the LTE network.

I'd love to make these changes available, but this would be my first commit to an open source project and I have no idea how the non-technical side of that goes, please bear with me.

I just cloned the repository, I guess I should make a branch, and then push to said branch. 

1) Is there a specific name I should name the branch or can I come up with my own? 
2) Do I need to register/request access somewhere?

Kind regards,
Govert.

**Comment by earlephilhower** [comment]
Please read the GH guide: https://docs.github.com/en/get-started/quickstart/contributing-to-projects

Basically you will **fork** this repo to your own account, then clone **your** fork, make a branch, do the changes locally, commit and push to **your** GH fork.  After that, the web interface will give you the option to open a pull request for your branch on this main repo.

As for the classes, is there not a library that they already provide for doing this (i.e. no need to edit the variant)?  Are they generally useful to others, etc.?

**Reply by Hoxolotl** [reply]
Good point, I hadn't looked it up. Going to have a go with https://github.com/SodaqMoja/Sodaq_R4X might be a suitable library to use.
